aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -221,8 +253,21 @@ Tensor& bernoulli_tensor_cuda_(Tensor &self, const Tensor& p_, Generator* gen) {
 
 Tensor& bernoulli_scalar_cuda_(Tensor &self, double p, Generator* gen) {
   AT_CHECK(0 <= p && p <= 1, "bernoulli_ expects p to be in [0, 1], but got p=", p);
+  auto gen_ = at::detail::checkGeneratorWithDefault(gen, kCUDA);
+  uint64_t block_size = 512; // AT_APPLY_THREADS_PER_BLOCK in CUDAApplyUtils.cuh
+  uint64_t step = 4;
+  uint64_t total_elements = self.numel();
+  // grid calculation from getApplyGrid() in CUDAApplyUtils.cuh
+  uint64_t grid_size = (total_elements + (block_size * step) - 1) / (block_size * step);
+  #if CUDA_VERSION < 9000
+    if (!self.is_contiguous()) {
+      uint64_t blocks_per_sm = 4; // AT_APPLY_BLOCKS_PER_SM in CUDAApplyUtils.cuh
```

Comment:
I wanted to add make it this function `unsigned int blocks_per_sm = at::cuda::getCurrentDeviceProperties()->maxThreadsPerMultiProcessor/block_size;` but CUDAApplyUtils.cuh currently hard codes blocks_per_sm and block_size and launches kernels with those hardcoded values. Hence, if the value of `at::cuda::getCurrentDeviceProperties()->maxThreadsPerMultiProcessor` doesn't result into a blocks_per_sm of 4, the offset calculation would be off and that's why hardcoded here. I could have modified CUDAApplyUtils.cuh but wanted to avoid doing it for this PR, but if you think it's necessary, I'll do it.